### ISSUE TITLE
Navigation breaks after 4.0.0-beta.23

### ISSUE THREAD

**Issue by thatjuan** [aksonov/react-native-router-flux#3202]
### Version

- react-native-router-flux v4.0.0 =
- react-native v0.56.0

### Expected behaviour

`Actions.SignIn();` should load up the SignIn Scene.

### Actual behaviour

Nothing happens. No error messages.

###  Issue
I upgraded from 4.0.0-beta31 to 4.0.0 and my app gets stuck in my initial scene. There's code in the main component of the initial scene that determines whether the user is logged in or not, and based on that they're redirected to a dashboard scene, or a log in scene. After the upgrade, the app just sits in the initial scene, and goes nowhere. 

I have tried:

1. Adding a button to the initial scene that when pressed logs something and then calls `Actions.SignIn(). The button logs but doesn't redirect.
2. Checking my Scene setup against the example app. I see nothing obvious, but I did try changing the structure of my scenes to all be under a Stack as opposed to being nested under a Scene. This change had no impact.
3.  Tried 4.0.1 and 4.0.2, with the same result. 4.0.0-beta40 fails with a red screen. 

My app is structured more or less like this:

```
<Provider store={store}>
    <ActionSheetProvider>
         <Router getSceneStyle={getSceneStyle}>
             <Stack key='root' hideNavBar panHandlers={null}>
                 <Scene key="app" component={App} title="App" initial/>

                 <Scene key='SignIn' component={SignIn} />

                 <Scene key='SignUp' component={SignUp}/>

                 <Scene key='GeneralSettings' component={GeneralSettings} />

                 <Scene key='Dashboard' component={Dashboard} type={ActionConst.POP_TO}/>
            </Stack>
         </Router>
    </ActionSheetProvider>
</Provider>
```

The App scene loads fine, but it can't navigate anywhere else. This all works fine on 4.0.0 beta 23, but breaks on any newer versions.

**Comment by daviscabral** [comment]
Could you reproduce the error using the `ReduxExample` from the project? If yes, do you mind sharing this - so I can check it and see what is going on?

**Comment by mikaelrosquist** [comment]
Had the exact same issue. Was able to "solve" it by setting a timeout before pushing the _SignIn_ scene.
`Actions.SignIn()` is not defined at first. Looks like a race condition.

**Comment by thatjuan** [comment]
@mikaelrosquist - I added `Actions.SignIn()` to the `onPress` of a button to rule that out. It never redirects, so we may be running into different issues.

**Comment by daviscabral** [comment]
I believe 4.0.0 is really broken, and that's why @aksonov released 4.0.1 right after. I tried to release 4.0.2 to fix a couple of other issues right after as well. But I will only have time to dedicate to code here in the weekend now. 

Anything that you guys might provide to help with that, specs, examples, would really make the testing/fixing process easier. Thank you.

**Comment by daviscabral** [comment]
I noticed that with the last release, a lot of issues might be happening due the wrong version match between a lot of dependencies.

Could you check if your version is compatible?

| `react-native` | `react`         |
|----------------|-----------------
| 0.46.x         | 16.0.0-alpha.12 |
| 0.47.x         | 16.0.0-alpha.12 |
| 0.48.x         | 16.0.0-alpha.12 |
| 0.49.x         | 16.0.0-beta.5   |
| 0.50.x         | 16.0.0          |
| 0.51.x         | 16.0.0          |
| 0.52.x         | 16.2.0          |
| 0.54.x         | 16.3.0-alpha.1  |
| 0.55.x         | 16.3.1          |
| 0.56.x         | 16.4.1          | 

Also, check the breaking changes related to the version you are using and the most recent - that might shed a light here:
https://github.com/react-native-community/react-native-releases/blob/master/CHANGELOG.md

If you think this was closed by mistake, please reopen the ticket after verifying the information above.
